object/slicer/options.go

```diff
@@ -61,6 +64,17 @@ func (x *Options) SetPayloadBuffer(payloadBuffer []byte) {
 	x.payloadBuffer = payloadBuffer
 }
 
+// SetPayloadSize allows to specify object's payload size known in advance. If
+// set, reading functions will read at least size bytes while writing functions
+// will expect exactly size bytes.
+//
+// If the size is known, the option is recommended as it improves the
+// performance of the application using the [Slicer].
+func (x *Options) SetPayloadSize(size uint64) {
```

Comment:
still do not get what this option solves. it is only used to return some `UnexpectedEOF` error if a user gave us more than he said. more options but what problems/UX experience does this fix?

Reply:
currently - more efficient payload buffer management

Reply:
seems like it was not true at the moment of the comment. but now it is ok